Common/Interfaces/IBusyCollection.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace QuantConnect.Interfaces
+{
+    /// <summary>
+    /// Interface used to handle items being processed and communicate busy state
+    /// </summary>
+    /// <typeparam name="T">The item type being processed</typeparam>
+    public interface IBusyCollection<T> : IDisposable
```

Comment:
Smart! Let's us test some of these complex/multi-threaded interactions in a _hopefully_ more deterministic way.